app/src/main/java/org/systers/mentorship/view/fragments/ResetPasswordFragment.kt

```diff
@@ -0,0 +1,78 @@
+package org.systers.mentorship.view.fragments
+
+import android.app.Dialog
+import android.arch.lifecycle.Observer
+import android.arch.lifecycle.ViewModelProviders
+import android.os.Bundle
+import android.support.v4.app.DialogFragment
+import android.support.v7.app.AlertDialog
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.Toast
+import kotlinx.android.synthetic.main.fragment_reset_password.view.*
+import org.systers.mentorship.R
+import org.systers.mentorship.remote.requests.ResetPassword
+import org.systers.mentorship.viewmodels.ResetPasswordViewModel
+
+/**
+ * The fragment is responsible for changing User password
+ */
+class ResetPasswordFragment : DialogFragment() {
+
+    companion object {
+        /**
+         * Creates an instance of ResetPasswordFragment
+         */
+        fun newInstance() = ResetPasswordFragment()
+    }
+
+    private lateinit var resetPasswordViewModel: ResetPasswordViewModel
+    private lateinit var resetDialogView: View
+    private lateinit var currentPassword: String
+    private lateinit var newPassword: String
+    private lateinit var confirmPassword: String
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        resetPasswordViewModel = ViewModelProviders.of(this).get(ResetPasswordViewModel::class.java)
+        resetPasswordViewModel.successfulUpdate.observe(this, Observer { successful ->
+
+            if (successful != null)
```

Comment:
```suggestion
            if (successful != null) {
                Toast.makeText(activity, resetPasswordViewModel.message, Toast.LENGTH_SHORT).show()
            }
            dismiss()
```
It looks weird, hard to understand that the toast only comes if the request is not successful.